### ISSUE TITLE
Code Freeze: Modularize and CLIify tasks

### ISSUE THREAD

**Issue by psealock** [woothemes/woocommerce#37353]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Create a Commander code freeze CLI tool that performs everything [our current code freeze action](https://github.com/woocommerce/woocommerce/blob/trunk/.github/workflows/release-code-freeze.yml) does. Break up the tasks into unique parts so they can run independently and be tested separately.

Once the tool is in place, replace our code freeze action with either one action that calls the CLI sequentially or multiple actions that call each part separately, which ever makes more sense. Keep in mind the following principles:

* handle failures gracefully
* ensure that when a failure happens, someone can easily know what was completed and manually finish the bits that didn't run
* easy to maintain
* steps should be testable
* cache where appropriate

- [x] https://github.com/woocommerce/woocommerce/issues/37902
- [x] https://github.com/woocommerce/woocommerce/issues/37903
- [x] https://github.com/woocommerce/woocommerce/issues/37904
- [x] https://github.com/woocommerce/woocommerce/issues/37905

**Comment by psealock** [comment]
Closing this in favour of a Project Thread
